### ISSUE TITLE
Can't override base class dependencies container with dependency built by callable

### ISSUE THREAD

**Issue by shaunc** [ets-labs/python-dependency-injector#374]
Here is another tricky one I stumbled on --
```python

from dependency_injector import providers, containers

class X(containers.DeclarativeContainer):
    foo = providers.Dependency(instance_of=str)

def build_x():
  return X(foo="1")

class A(containers.DeclarativeContainer):

  x = providers.DependenciesContainer(**X.providers)
  y = x.foo

@containers.copy(A)
class B1(A):
  some_x = providers.Callable(build_x)
  x = some_x.provided

b1 = B1()
print(b1.y())  # fails with "Dependency is not defined"
```

I also can't get it to work by repeating x definition and overriding:
```python
@containers.copy(A)
class B2(A):
  some_x = providers.Callable(build_x)
  x = providers.DependenciesContainer(**X.providers)
  x.override(some_x.provided)
```
Gives `TypeError: 'builtin_function_or_method' object is not iterable` on module load.

**Comment by rmk135** [comment]
Hey @shaunc ,

Here is a way to make it work:

```python
from dependency_injector import providers, containers

class X(containers.DeclarativeContainer):
  foo = providers.Dependency(instance_of=str)

def build_x():
  return X(foo="1")

class A(containers.DeclarativeContainer):

  x = providers.DependenciesContainer(**X.providers)
  y = x.foo

@containers.copy(A)
class B2(A):
  some_x = providers.Container(build_x)  # <-- Replace Callable with Container
  x = providers.DependenciesContainer(**X.providers)
  x.override(some_x)  # <-- Remove .provided

b1 = B2()
print(b1.y())  # fails with "Dependency is not defined"
```

PS: My apologies it took so long to answer. Was overwhelmed with other stuff.

**Comment by rmk135** [comment]
I'm not quite sure why 1st example doesn't work. I'll double-check it.

**Comment by rmk135** [comment]
I have fixed an issue in ``@containers.copy``. Now 1st example also works with little modifications:

```python
from dependency_injector import providers, containers

class X(containers.DeclarativeContainer):
    foo = providers.Dependency(instance_of=str)

def build_x():
    return X(foo="1")

class A(containers.DeclarativeContainer):

    x = providers.DependenciesContainer(**X.providers)
    y = x.foo

@containers.copy(A)
class B1(A):
    some_x = providers.Container(build_x)  # <-- Replace Callable with Container
    x = some_x  # <-- Remove .provided

b1 = B1()
print(b1.y())  # prints "1"
```

**Comment by rmk135** [comment]
Fixed in ``4.11.1``. It's been publishing on PyPI right now and will be ready as soon as https://github.com/ets-labs/python-dependency-injector/actions/runs/515305692 is done. Btw, it's a new CD pipeline on Github Actions.

**Comment by shaunc** [comment]
Great -- thank you!
